Fix version number according to PEP440

abhiabhi94 opened the pull request:
- version number is now a part of the package and can be accessed by `comment.__version__`.
- use this version when adding javascript urls for translation to improve performance. See: https://docs.djangoproject.com/en/dev/topics/i18n/translation/#note-on-performance

abhiabhi94 commented:
[PEP440](https://www.python.org/dev/peps/pep-0440/)

[TLDR](https://stackoverflow.com/a/27494005/5317267)(too long didn't read)

Radi85 commented on comment/__init__.py:
```diff
@@ -1 +1,8 @@
 default_app_config = 'comment.apps.CommentConfig'
+
+__version__ = "2.0.0"
```
Why didn't you just read VERSION file here instead of using a static string. I would prefer to change VERSION file on each releasing rather than changing the __init__ file.

abhiabhi94 replied:
I had not seen the manifest file when writing this. I thought we are not shipping with the package to the end users.

Now that you put it this way, this solution seems really silly :(

I actually looked for this the source code of *pip*.

I don't know why they do it like this? Maybe because they don't want users to download one extra file or something else?

Shall I revert the change?

Radi85 replied:
Yes please.

Radi85 commented on comment/__init__.py:
```diff
@@ -1 +1,8 @@
+__version__ = "2.0.0"
+
+
+def _get_version():
```
Why do you use `_` prefix in the function name since you are using it outside the module?

abhiabhi94 replied:
I have read in Python(also in Django) `_` is also used to show an internal function(or class or attribute or simply an object) which might be changed in the future without any notice.

Radi85 replied:
I am not sure about that but what I know is that in python the underscore before the function or the variable means that this member is private and shouldn't be used outside the scope.

abhiabhi94 replied:
To quote the official `django` documentation:

> **APIs marked as internal**
   Certain APIs are explicitly marked as “internal” in a couple of ways:
    Some documentation refers to internals and mentions them as such. If the documentation says that something is internal, we reserve the right to change it.
    Functions, methods, and other objects prefixed by a leading underscore (_). This is the standard Python way of indicating that something is private; if any method starts with a single _, it’s an internal API.


Reference: https://docs.djangoproject.com/en/3.1/misc/api-stability/#apis-marked-as-internal

Radi85 commented on docs/source/conf.py:
```diff
@@ -19,23 +19,33 @@
-    with open(os.path.join(version_file_dir, 'VERSION')) as version_file:
-        ver = version_file.read().strip()
-    return ver
+docs_dir = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
```
To be more specific and explicit 
```suggestion
docs_parent_dir = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
```

Radi85 commented on docs/source/conf.py:
```diff
@@ -19,23 +19,33 @@
+import re
 
 
-def get_version():
```
I would say this method was more cleaner.

Radi85 reviewed (changes requested):
I would say removing VERSION file led to write complex code to extract the version number from __init__.py file. I would prefer to stick with the old approach for getting the version from `VERSION` file and just keep the changes that are required for JS i18n and comment.__version__

abhiabhi94 committed:
Fix version number according to PEP440

- version number is now a part of the package and can be accessed by `comment.__version__`.
- use this version when adding javascript urls for translation to improve performance.

Radi85 commented on comment/tests/test_version.py:
```diff
@@ -0,0 +1,19 @@
+
+class TestGetVersion(TestCase):
+    _parent_project_dir = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+    version_file = os.path.join(_parent_project_dir, 'VERSION')
```
Since you are mocking theversion value below why would you read the real value from the file? the variable `version_file` is unused.
In my opinion it would be more realistic if you assert the `_get_version()` with the actual value in `version_file`.

abhiabhi94 replied:
No, the actual value will keep changing. So the test will also require a change every time there's a change of version.

Regarding the variables, I was using them for debugging something. Forgot to remove. will remove them if everything else looks good to you.

Radi85 replied:
> No, the actual value will keep changing. So the test will also require a change every time there's a change of version.

the value is read from the VERSION file so if you compare `version_file.read().strip()` with `_get_version()` will do the trick and will not require to change anything when the version is incremented.

However, feel free to use either way!

abhiabhi94 replied:
> the value is read from the VERSION file so if you compare version_file.read().strip() with _get_version() will do the trick

It will not be in every case. Suppose I mock the contents to be `v2.0.0`. The function `_get_version()` would return `2.0.0` which would not be the same as contents of the file.

hossein-shahsavan commented:
hi how can i use persian calender instead of the deafult calender for comments???

hossein-shahsavan commented:
for changing the calender i must change the comment model but i dont have access
